### PR TITLE
Add rules for FluxCD alerting

### PR DESCRIPTION
**Pull request by kubasobon**
Towards https://github.com/giantswarm/giantswarm/issues/19449 and https://github.com/giantswarm/giantswarm/issues/19455

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Ops recipes mentioned in the rules must exist https://github.com/giantswarm/giantswarm/pull/19699
- [x] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).
- [x] Alerting rules must have a comment documenting why it needs to exist.

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/fluxcd.rules.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  creationTimestamp: null
+  labels:
+    {{- include "labels.common" . | nindent 4 }}
+  name: fluxcd.rules
+  namespace: {{ .Values.namespace  }}
+spec:
+  groups:
+  - name: fluxcd
+    rules:
+    - alert: FluxCDHelmReleaseFailed
```
Comment (MarcelMue) [inline helm/prometheus-rules/templates/alerting-rules/fluxcd.rules.yml:13]: Can we rename `fluxCD` to `flux` here? Should still be obvious what it is :D

File: helm/prometheus-rules/templates/alerting-rules/fluxcd.rules.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      annotations:
+        description: |-
+          {{`FluxCD HelmRelease on {{ $labels.installation }}/{{ $labels.cluster_id }} in {{ $labels.namespace }}/{{ $labels.name }} is stuck in Failed state.`}}
+        opsrecipe: fluxcd/failing
+      expr: gotk_reconcile_condition{type="Ready", status="False", kind="HelmRelease"}
+      for: 10m
+      labels:
+        area: kaas
```
Comment (MarcelMue) [inline helm/prometheus-rules/templates/alerting-rules/fluxcd.rules.yml:21]: Is this actually kaas?

Comment (kubasobon) [reply helm/prometheus-rules/templates/alerting-rules/fluxcd.rules.yml:21]: I based the area on ArgoCD alerts. Does `managedservices` better?

File: helm/prometheus-rules/templates/alerting-rules/flux.rules.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  namespace: {{ .Values.namespace  }}
+spec:
+  groups:
+  - name: fluxcd
+    rules:
+    - alert: FluxHelmReleaseFailed
+      annotations:
+        description: |-
+          {{`Flux HelmRelease on {{ $labels.installation }}/{{ $labels.cluster_id }} in {{ $labels.namespace }}/{{ $labels.name }} is stuck in Failed state.`}}
+        opsrecipe: fluxcd/failing
```
Comment (rossf7) [inline helm/prometheus-rules/templates/alerting-rules/flux.rules.yml:17]: I added an ops recipe for helmrelease problems. Can we link to that somehow?

Comment (rossf7) [reply helm/prometheus-rules/templates/alerting-rules/flux.rules.yml:17]: Adding the link would help 🤦‍♂️ here it is

https://intranet.giantswarm.io/docs/support-and-ops/ops-recipes/fluxcd-helmrelease-not-ready/

Comment (kubasobon) [reply helm/prometheus-rules/templates/alerting-rules/flux.rules.yml:17]: I've just seen it, it's so helpful! I'll redirect alert for `HelmReleases` to your recipe.

**Review by MarcelMue — APPROVED** [review]
Works for me - I have no idea about areas in the new team structure, let's go

File: helm/prometheus-rules/templates/alerting-rules/flux.rules.yml (hunk 3 above)
Comment (rossf7) [reply helm/prometheus-rules/templates/alerting-rules/flux.rules.yml]: Thx!

**Review by rossf7 — APPROVED** [review]

**Pull request merged**